theano/gof/op.py

```diff
@@ -783,6 +783,9 @@ def __new__(cls, *args, **kwargs):
         # this function exists to silently and transparently ensure that all
         # existing Ops get a _op_use_c_code attribute
         obj = object.__new__(cls)
+        # Ensuring all the Ops created henceforth has __props__
+        if not hasattr(obj, '__props__'):
+            raise
```

Comment:
I don't know if i should keep this. This raises a lot of error, regarding few Ops in old backend and tests that don't implement `__props__`. Should i keep these two lines and add the `__props__` to the required places ?
I thought this check would be useful because, if there is an Op that is being created in future in theano, this would ensure that it has `__props__`.